src/core/prometheus.cc

```diff
@@ -712,23 +723,46 @@ future<> write_text_representation(output_stream<char>& out, const config& ctx,
     });
 }
 
-future<> write_protobuf_representation(output_stream<char>& out, const config& ctx, metric_family_range& m) {
-    return do_for_each(m, [&ctx, &out](metric_family& metric_family) mutable {
+future<> write_protobuf_representation(output_stream<char>& out, const config& ctx, metric_family_range& m, std::function<bool(const mi::labels_type&)> filter) {
+    return do_for_each(m, [&ctx, &out, filter](metric_family& metric_family) mutable {
         std::string s;
         google::protobuf::io::StringOutputStream os(&s);
-
+        metric_aggregate_by_labels aggregated_values(metric_family.metadata().aggregate_labels);
+        bool should_aggregate = !metric_family.metadata().aggregate_labels.empty();
         auto& name = metric_family.name();
         pm::MetricFamily mtf;
-
+        bool empty_metric = true;
         mtf.set_name(ctx.prefix + "_" + name);
         mtf.mutable_metric()->Reserve(metric_family.size());
-        metric_family.foreach_metric([&mtf, &ctx](auto value, auto value_info) {
-            fill_metric(mtf, value, value_info.id, ctx);
+        metric_family.foreach_metric([&mtf, &ctx, &name, &filter, &aggregated_values, &empty_metric, should_aggregate](auto value, auto value_info) {
+            fill_metric(mtf, value, value_info.id.labels(), ctx);
```

Comment:
shall we skip this `fill_metric()` call based on the policies used in the next line? in other words, shall we just drop this line?

Reply:
it was a rebase problem, it shouldn't have been left there, and will be removed